flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -407,8 +407,9 @@ public RunnableFuture<SnapshotResult<KeyedStateHandle>> snapshot(
                                         metrics.reportSnapshotResult(snapshotResult))
                         .thenApply(
                                 snapshotResult ->
-                                        SnapshotResult.of(
-                                                snapshotResult.getJobManagerOwnedSnapshot())));
+                                        SnapshotResult.withLocalState(
```

Comment:
There are "@Nonnull" annotations in SnapshotResult#withLocalState.
So I think it's better to adding some check before calling it.